File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/ui/dialogs/DialogConsumePotionOfClumsiness.java
package it.paranoidsquirrels.idleguildmaster.ui.dialogs;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.core.content.res.ResourcesCompat;
import androidx.viewbinding.ViewBinding;
import it.paranoidsquirrels.idleguildmaster.MainActivity;
import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.UIUtils;
import it.paranoidsquirrels.idleguildmaster.Utils;
import it.paranoidsquirrels.idleguildmaster.databinding.DialogConsumePotionOfClumsinessBinding;
import it.paranoidsquirrels.idleguildmaster.databinding.LayoutAdventurerChangeTraitBinding;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.adventurers.Adventurer;
import it.paranoidsquirrels.idleguildmaster.storage.data.items.Item;

/* JADX INFO: loaded from: classes3.dex */
public class DialogConsumePotionOfClumsiness extends CustomDialog {
    private DialogConsumePotionOfClumsinessBinding binding;
    private AlertDialog confirmDialog;

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected ViewBinding getBinding() {
        return this.binding;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected void setBinding(ViewBinding viewBinding) {
        this.binding = (DialogConsumePotionOfClumsinessBinding) viewBinding;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected String getTitle() {
        return getString(R.string.dialog_consume_evo23_title);
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected ViewBinding inflate(LayoutInflater layoutInflater, ViewGroup viewGroup, boolean z) {
        DialogConsumePotionOfClumsinessBinding dialogConsumePotionOfClumsinessBindingInflate = DialogConsumePotionOfClumsinessBinding.inflate(layoutInflater, viewGroup, z);
        this.binding = dialogConsumePotionOfClumsinessBindingInflate;
        return dialogConsumePotionOfClumsinessBindingInflate;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected void initialize(Bundle bundle) {
        this.binding.list.removeAllViews();
        for (final Adventurer adventurer : MainActivity.data.getAdventurers()) {
            if (adventurer.getPotionsDrank().get(10) != 0) {
                LayoutAdventurerChangeTraitBinding layoutAdventurerChangeTraitBindingInflate = LayoutAdventurerChangeTraitBinding.inflate(getLayoutInflater(), this.binding.list, false);
                if (adventurer.isAscended()) {
                    layoutAdventurerChangeTraitBindingInflate.containerAdventurer.setBackgroundResource(R.drawable.object_border_ascended);
                    layoutAdventurerChangeTraitBindingInflate.image.setBackgroundResource(R.drawable.object_border_rounded_left_ascended);
                    layoutAdventurerChangeTraitBindingInflate.name.setTextColor(getResources().getColor(R.color.ascended_unit, getContext().getTheme()));
                }
                layoutAdventurerChangeTraitBindingInflate.image.setImageDrawable(ResourcesCompat.getDrawable(getResources(), adventurer.getImageId(), getContext().getTheme()));
                layoutAdventurerChangeTraitBindingInflate.level.setText(String.valueOf(adventurer.getLevel()));
                layoutAdventurerChangeTraitBindingInflate.cardView.setVisibility(adventurer.getDoctrine().getTrueClass().equals("EmptyDoctrine") ? 8 : 0);
                layoutAdventurerChangeTraitBindingInflate.doctrine.setImageDrawable(ResourcesCompat.getDrawable(getResources(), adventurer.getDoctrine().getIdImage(), getContext().getTheme()));
                layoutAdventurerChangeTraitBindingInflate.name.setText(getString(adventurer.getIdName()));
                layoutAdventurerChangeTraitBindingInflate.traits.setText(UIUtils.traitsToShortString(adventurer, getResources()));
                layoutAdventurerChangeTraitBindingInflate.getRoot().setOnClickListener(new View.OnClickListener() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogConsumePotionOfClumsiness$$ExternalSyntheticLambda0
                    @Override // android.view.View.OnClickListener
                    public final void onClick(View view) {
                        DialogConsumePotionOfClumsiness.this.m255x13729ce5(adventurer, view);
                    }
                });
                this.binding.list.addView(layoutAdventurerChangeTraitBindingInflate.getRoot());
            }
        }
    }

    /* JADX INFO: renamed from: lambda$initialize$2$it-paranoidsquirrels-idleguildmaster-ui-dialogs-DialogConsumePotionOfClumsiness, reason: not valid java name */
    /* synthetic */ void m255x13729ce5(final Adventurer adventurer, View view) {
        if (this.confirmDialog != null) {
            return;
        }
        AlertDialog actionDialog = UIUtils.getActionDialog(getContext(), Integer.valueOf(R.string.confirm), String.format(getString(R.string.dialog_clumsiness_confirm), getString(adventurer.getIdName())), R.string.yes, new DialogInterface.OnClickListener() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogConsumePotionOfClumsiness$$ExternalSyntheticLambda1
            @Override // android.content.DialogInterface.OnClickListener
            public final void onClick(DialogInterface dialogInterface, int i) {
                DialogConsumePotionOfClumsiness.this.m253x527ff0e3(adventurer, dialogInterface, i);
            }
        });
        this.confirmDialog = actionDialog;
        actionDialog.setOnDismissListener(new DialogInterface.OnDismissListener() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogConsumePotionOfClumsiness$$ExternalSyntheticLambda2
            @Override // android.content.DialogInterface.OnDismissListener
            public final void onDismiss(DialogInterface dialogInterface) {
                DialogConsumePotionOfClumsiness.this.m254x32f946e4(dialogInterface);
            }
        });
        this.confirmDialog.show();
    }

    /* JADX INFO: renamed from: lambda$initialize$0$it-paranoidsquirrels-idleguildmaster-ui-dialogs-DialogConsumePotionOfClumsiness, reason: not valid java name */
    /* synthetic */ void m253x527ff0e3(Adventurer adventurer, DialogInterface dialogInterface, int i) {
        Utils.removeItemFromStorage(Item.getInstance("PotionOfClumsiness", 1));
        adventurer.getPotionsDrank().resetAgility();
        if (MainActivity.shownDialogItemDetail != null) {
            MainActivity.shownDialogItemDetail.initialize(null);
        }
        if (MainActivity.shownDialogStorage != null) {
            MainActivity.shownDialogStorage.update();
        }
        dismiss();
    }

    /* JADX INFO: renamed from: lambda$initialize$1$it-paranoidsquirrels-idleguildmaster-ui-dialogs-DialogConsumePotionOfClumsiness, reason: not valid java name */
    /* synthetic */ void m254x32f946e4(DialogInterface dialogInterface) {
        this.confirmDialog = null;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected void attachListeners() {
        this.binding.close.setOnClickListener(new View.OnClickListener() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogConsumePotionOfClumsiness$$ExternalSyntheticLambda3
            @Override // android.view.View.OnClickListener
            public final void onClick(View view) {
                DialogConsumePotionOfClumsiness.this.m252x8f2820f6(view);
            }
        });
    }

    /* JADX INFO: renamed from: lambda$attachListeners$3$it-paranoidsquirrels-idleguildmaster-ui-dialogs-DialogConsumePotionOfClumsiness, reason: not valid java name */
    /* synthetic */ void m252x8f2820f6(View view) {
        dismiss();
    }

    @Override // androidx.fragment.app.DialogFragment, androidx.fragment.app.Fragment
    public void onStart() {
        super.onStart();
        MainActivity.shownDialogConsumePotionOfClumsiness = this;
    }

    @Override // androidx.fragment.app.DialogFragment, androidx.fragment.app.Fragment
    public void onStop() {
        MainActivity.shownDialogConsumePotionOfClumsiness = null;
        super.onStop();
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/ui/dialogs/DialogConsumePotionOfClumsiness.kt
package it.paranoidsquirrels.idleguildmaster.ui.dialogs

import android.app.AlertDialog
import android.os.Bundle
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.content.res.ResourcesCompat
import androidx.viewbinding.ViewBinding
import it.paranoidsquirrels.idleguildmaster.MainActivity
import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.UIUtils
import it.paranoidsquirrels.idleguildmaster.Utils
import it.paranoidsquirrels.idleguildmaster.databinding.DialogConsumePotionOfClumsinessBinding
import it.paranoidsquirrels.idleguildmaster.databinding.LayoutAdventurerChangeTraitBinding
import it.paranoidsquirrels.idleguildmaster.storage.data.items.Item

class DialogConsumePotionOfClumsiness : CustomDialog() {
    private var binding: DialogConsumePotionOfClumsinessBinding? = null
    private var confirmDialog: AlertDialog? = null

    override fun getBinding(): ViewBinding = binding!!

    override fun setBinding(viewBinding: ViewBinding) {
        binding = viewBinding as DialogConsumePotionOfClumsinessBinding
    }

    override fun getTitle(): String = getString(R.string.dialog_consume_evo23_title)

    override fun inflate(inflater: LayoutInflater, container: ViewGroup?, attachToRoot: Boolean): ViewBinding {
        val b = DialogConsumePotionOfClumsinessBinding.inflate(inflater, container, attachToRoot)
        binding = b
        return b
    }

    override fun initialize(arguments: Bundle?) {
        val b = binding ?: return
        b.list.removeAllViews()
        for (adventurer in MainActivity.data.adventurers) {
            if ((adventurer.potionsDrank?.get(10) ?: 0) != 0) {
                val itemBinding = LayoutAdventurerChangeTraitBinding.inflate(layoutInflater, b.list, false)
                if (adventurer.isAscended()) {
                    itemBinding.containerAdventurer.setBackgroundResource(R.drawable.object_border_ascended)
                    itemBinding.image.setBackgroundResource(R.drawable.object_border_rounded_left_ascended)
                    context?.let { ctx ->
                        itemBinding.name.setTextColor(resources.getColor(R.color.ascended_unit, ctx.theme))
                    }
                }
                context?.let { ctx ->
                    itemBinding.image.setImageDrawable(ResourcesCompat.getDrawable(resources, adventurer.imageId, ctx.theme))
                    adventurer.doctrine?.let { doc ->
                        itemBinding.doctrine.setImageDrawable(ResourcesCompat.getDrawable(resources, doc.idImage, ctx.theme))
                    }
                }
                itemBinding.level.text = adventurer.level.toString()
                itemBinding.cardView.visibility = if (adventurer.doctrine?.trueClass == "EmptyDoctrine") 8 else 0
                itemBinding.name.text = getString(adventurer.idName)
                itemBinding.traits.text = UIUtils.traitsToShortString(adventurer, resources)
                itemBinding.root.setOnClickListener {
                    if (confirmDialog != null) return@setOnClickListener
                    val actionDialog = UIUtils.getActionDialog(
                        context,
                        R.string.confirm,
                        String.format(getString(R.string.dialog_clumsiness_confirm), getString(adventurer.idName)),
                        R.string.yes
                    ) { _, _ ->
                        Utils.removeItemFromStorage(Item.getInstance("PotionOfClumsiness", 1))
                        adventurer.potionsDrank?.resetAgility()
                        MainActivity.shownDialogItemDetail?.initialize(null)
                        MainActivity.shownDialogStorage?.update()
                        dismiss()
                    }
                    confirmDialog = actionDialog
                    actionDialog.setOnDismissListener {
                        confirmDialog = null
                    }
                    confirmDialog?.show()
                }
                b.list.addView(itemBinding.root)
            }
        }
    }

    override fun attachListeners() {
        binding?.close?.setOnClickListener {
            dismiss()
        }
    }

    override fun onStart() {
        super.onStart()
        MainActivity.shownDialogConsumePotionOfClumsiness = this
    }

    override fun onStop() {
        MainActivity.shownDialogConsumePotionOfClumsiness = null
        super.onStop()
    }
}
